### ISSUE TITLE
Disable random slug generation for containers

### ISSUE THREAD

**Issue by deiu** [solid/solid#42]
During a POST request (with a Slug header set), the server may decide to create a random string that gets appended to the Slug value if a resource with the same name already exists. While this feature is really useful for creating random resources, it seems to be very annoying when applied to creation of containers.

I personally feel that most applications and users care a lot about the names of containers, and less so for resource names (which can sometimes be random).

**Comment by melvincarvalho** [comment]
+1 ran into this issue also ... but can live with the way things are also

**Comment by dmitrizagidulin** [comment]
+1, running into this with test resource creation and cleanup on ldnode.

**Comment by sandhawke** [comment]
I read your proposal as conflicting with the LDP Rec.   Some ways that come to my mind to address that are: 

- use a header other than slug

- use PUT instead of POST to create the container, wiith If-None-Match: * to prevent accidental overwrite

- document the thing you're POSTing to as having a different kind of handling of slugs

I lean toward PUT as the simplest.

**Comment by bblfish** [comment]
`PUT` seems good. But I think it should go with a new type of container such as what I called [iContainer]( https://www.w3.org/2012/ldp/track/issues/50) for Intuitive Container.  The current containers don't allow you to know where the resource will be created. I am willing to be that most of our servers's containers are `iContainers`.  

There is a way of doing `PUT` and specifying that you don't want to overwrite whatever is there which is what you want.

**Comment by sandhawke** [comment]
Yes, agreed, with PUT it needs to be a different type of container controlling the URL space in which you PUT.   Years ago, Tim and Andrei and I called this HierarchicalContainer, but now I'm suggesting in #37 we call it "Folder", in part because the term "container" has been commandeered via LXC.

And yes, in using PUT for this, one needs to use "If-None-Match: *" to avoid accidentally over-writing whatever might already be there.

**Comment by bblfish** [comment]
yes, `Folder` is a good name actually.

**Comment by dmitrizagidulin** [comment]
This is still an issue -- we currently don't support container creation via PUT (on either Gold or LDNode, and we mention in the spec that it's not supported).

**Comment by Mitzi-Laszlo** [comment]
You can find more conversation on this subject here https://forum.solidproject.org/t/disable-random-slug-generation-for-containers/384
